File: VB/CustomAppointmentFormController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.UI

Namespace SchedulerInheritedAppointmentForm
	Public Class CustomAppointmentFormController
		Inherits AppointmentFormController
		Public Property Price() As Decimal
			Get
				Dim val As Object = EditedAppointmentCopy.CustomFields("Field1")

				If val Is Nothing OrElse val Is DBNull.Value Then
					Return 0
				Else
					Return Convert.ToDecimal(val)
				End If
			End Get
			Set(ByVal value As Decimal)
				EditedAppointmentCopy.CustomFields("Field1") = value
			End Set
		End Property

		Private Property SourcePrice() As Decimal
			Get
				Return CDec(SourceAppointment.CustomFields("Field1"))
			End Get
			Set(ByVal value As Decimal)
				SourceAppointment.CustomFields("Field1") = value
			End Set
		End Property

		Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment)
			MyBase.New(control, apt)
		End Sub

		Public Overrides Function IsAppointmentChanged() As Boolean
			If MyBase.IsAppointmentChanged() Then
				Return True
			End If
			Return SourcePrice <> Price
		End Function

		Protected Overrides Sub ApplyCustomFieldsValues()
			SourcePrice = Price
		End Sub
	End Class
End Namespace

File: CS/CustomAppointmentFormController.cs
using System;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.UI;

namespace SchedulerInheritedAppointmentForm {
    public class CustomAppointmentFormController : AppointmentFormController {
        public decimal Price {
            get {
                object val = EditedAppointmentCopy.CustomFields["Field1"];

                if (val == null || val == DBNull.Value)
                    return 0;
                else
                    return Convert.ToDecimal(val);
            }
            set {
                EditedAppointmentCopy.CustomFields["Field1"] = value;
            }
        }

        private decimal SourcePrice {
            get { return (decimal)SourceAppointment.CustomFields["Field1"]; }
            set { SourceAppointment.CustomFields["Field1"] = value; }
        }

        public CustomAppointmentFormController(SchedulerControl control, Appointment apt)
            : base(control, apt) {
        }

        public override bool IsAppointmentChanged() {
            if (base.IsAppointmentChanged())
                return true;
            return SourcePrice != Price;
        }

        protected override void ApplyCustomFieldsValues() {
            SourcePrice = Price;
        }
    }
}
